configure.ac

```diff
@@ -140,6 +141,9 @@ AM_CONDITIONAL(CHECK_TWO_NODE, test "$HAVE_cmap" = "1")
 AC_DEFINE_UNQUOTED(CHECK_VOTEQUORUM_HANDLE, $HAVE_votequorum, Turn on periodic checking of votequorum-handle)
 AM_CONDITIONAL(CHECK_VOTEQUORUM_HANDLE, test "$HAVE_votequorum" = "1")
 
+AC_DEFINE_UNQUOTED(USE_PACEMAKERD_API, $HAVE_pacemakerd_api, Turn on synchronization between sbd & pacemakerd)
+AM_CONDITIONAL(USE_PACEMAKERD_API, test "$HAVE_pacemakerd_api" = "1")
```

Comment:
I see an issue if we want both sbd and pacemaker to be able to handle any version of the other.

If newer sbd is built with older pacemaker, it won't ping. If pacemaker is later upgraded, pacemaker will see the newer sbd and wait for a ping that never comes (unless sbd is rebuilt). Similarly in the flip situation.

Since sbd links against pacemaker, we could bump the library version on the pacemaker side to ensure sbd must be rebuilt -- but we don't want an incompatible library bump in pacemaker at the moment, and it wouldn't be polite to require all pacemaker clients to rebuild when only sbd needs to. :-/

Another possibility would be having sbd use dynamic library functions to check whether the API function is available, but I'm not sure we want to introduce that complexity into sbd. And that wouldn't work on the pacemaker side since there's no library being linked there.

Distros can easily ensure they distribute compatible versions, but it's a tougher problem for users who build from source.

Reply:
As already written on the pacemaker-side of the issue I don't see a really good way to do that without the danger of introducing something that makes everything look as if was synced the new way but it actually isn't.
At least a mismatch in both ways keeps resources down like this.
What I could imagine is to make the pacemakerd-api conditional on the pacemaker-side. So sbdy building a new pacemaker and not wanting to get new sbd and build it against the new pacemaker can disable it. Still ugly ...